core/ledger/kvledger/txmgmt/statedb/statecouchdb/statecouchdb.go

```diff
@@ -721,6 +722,28 @@ func (vdb *VersionedDB) Close() {
 	// no need to close db since a shared couch instance is used
 }
 
+// Remove removes the couch dbs and redologger db for the channel.
+// It is not an error if a database does not exist.
+func (vdb *VersionedDB) Remove() error {
+	logger.Infof("Dropping CouchDB databases for channel %s", vdb.chainName)
+	metadataDBName := constructMetadataDBName(vdb.chainName)
+	for _, dbInfo := range vdb.channelMetadata.NamespaceDBsInfo {
+		// do not drop metadataDB until all other dbs are dropped
+		if dbInfo.DBName == metadataDBName {
+			continue
+		}
+		if _, err := dropDB(vdb.couchInstance, dbInfo.DBName); err != nil && !strings.Contains(err.Error(), "not_found") {
```

Comment:
Rather then looking at "not_found" in the error string at this level, it will be better to enhance `dropDB`. Does couch returns a specific http code for this case (I guess, 404). It will be more robust to check for code and let `dropDB` not return an error.